spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -372,12 +373,15 @@ def main(argv: Sequence[str]):
 
     file_out = os.path.abspath(arguments.o)
     append = bool(arguments.append)
-    if arguments.vert is not None:
-        levels = arguments.vert
-        fname_vert_level = arguments.vertfile
-    else:
-        levels = []
-        fname_vert_level = None
+    levels = arguments.vert
+    fname_vert_level = arguments.vertfile
+    if not os.path.isfile(fname_vert_level):
+        logger.warning(f"Vertebral level file {fname_vert_level} does not exist. Vert level information will "
+                       f"not be displayed. To use vertebral level information, you may need to run "
+                       f"`sct_warp_template` to generate the appropriate level file in your working directory.")
+        fname_vert_level = None  # Discard the default '-vertfile', so that we don't attempt to find vertebral levels
+        if levels:
+            raise FileNotFoundError("The vertebral level file must exist to use `-vert` to group by vertebral level.")
```

Comment:
Just to provide a little more context for why this change was made:

* The `-vertfile` argument has a default value of `./label/template/PAM50_levels.nii.gz`. (This corresponds to a path that gets created when the user has run `sct_warp_template`. Otherwise, this file will probably not exist.)
* Before this PR, to get around the issue of the default filepath potentially being nonexistent, we made the assumption that if the user has manually specified the related option, `-vert`, then they are signalling intent, and will have probably run `sct_warp_template` beforehand.
* So, `-vertfile` was only used if if `-vert` was also passed. (If `-vert` was not specified, then `-vertfile` would be overridden by `None`.) 
* However, this PR requires us to access the value of `-vertfile` even in cases where `-vert` isn't specified.
* This means we need a new way (apart from checking `-vert`) to ensure that we're not trying to access a nonexistent file.
* So, the new method directly checks for the existence of `-vertfile`, and sets it to `None` (and provides helpful warnings/errors) in the case where it doesn't exist.